### PR TITLE
Appointments history processing - closes #10

### LINKED ISSUE #10: Clean appointment history
**Scenario:**
As a developer I need to efficiently manage job appointment histories received from the PeopleXD API. These histories often contain raw, unprocessed data with contiguous appointments that have identical job and department values. Our objective is to simplify this data by merging these identical appointments into a unified form. Furthermore, the cleaned-up data should be returned as an ordered list with job and department titles for easier consumption and display in our system.

**User Story:**
As a user of the Employee Management System, I want the application to intelligently process and clean the job appointment history so that I can understand my career progression easily. The system should merge consecutive appointments that have the same job and department details into a single entry. Additionally, the cleaned history should be provided in an ordered list based on the appointment dates. This will allow me to have a clear and concise view of the job history without redundant or repetitive entries.

**Acceptance Criteria:**
1. The function should accept raw job appointment history data from the PeopleXD API.
2. Identical contiguous appointments (same job and department) should be merged into a single entry.
3. Each entry in the resulting list should contain job and department titles clearly.
4. The final output should be an ordered list of appointments based on start dates.
5. Unit tests should verify the merging functionality and correct ordering of the appointments.
6. The function should handle edge cases, such as overlapping appointments or empty input data, gracefully.

**Technical Details:**
- Input: Raw appointment data structure from PeopleXD API.
- Output: Cleaned and ordered list of appointments with job and department titles.

### PR DESCRIPTION
**Pull request by eoin-kilfeather**
This PR adds the `cleanAppointments(staff_no)` method to give a cleaned and processed appointment history more suitable for reuse in other systems.

It refactors the client into a set of services for ease of maintenance and testing.

It also adds a more descriptive `README.md` file and a `dummy_proj/main.js` file which allows a simple functional test.

### CODE REVIEW

File: .devcontainer/devcontainer.env.EXAMPLE [hunk 1]
```diff
@@ -0,0 +1,4 @@
+PEOPLEXD_URL=https://api.corehr.com/ws/<your_tenant_id>/corehr/
+PRODUCTION_PXD_SECRET=<your_secret>
+PRODUCTION_PXD_ID=<your_id>
+TEST_STAFF_ID=058627 # used in dummy_proj/main.js
```
Comment (cian-mcgovern) [inline .devcontainer/devcontainer.env.EXAMPLE]: Should you remove the staff id?

File: README.md [hunk 2]
```diff
@@ -1,14 +1,157 @@
...
+
+### Development Container
+
+This project includes a devcontainer configuration for Visual Studio Code, which provides a consistent development environment with all necessary dependencies. In includes a sample `devcontainer.env.EXAMPLE` file which you will need to rename `devcontainer.env` and populate with required values for your PeopleXD tenancy.
+
+```
+PEOPLEXD_URL=https://api.corehr.com/ws/<your_tenant_id>/corehr/
+PRODUCTION_PXD_SECRET=<your_secret>
+PRODUCTION_PXD_ID=<your_id>
+TEST_STAFF_ID=058627 # used in dummy_proj/main.js
```
Comment (cian-mcgovern) [inline README.md]: same as above

File: .devcontainer/devcontainer.env.EXAMPLE (hunk 1 above)
Comment (eoin-kilfeather) [reply .devcontainer/devcontainer.env.EXAMPLE]: okay

File: README.md (hunk 2 above)
Comment (eoin-kilfeather) [reply README.md]: done
